website/content/en/docs/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,396 @@
+
+Copy over the API spec and status from the SDK project's `pkg/apis/<group>/<version>/<kind>_types.go` to Kubebuilder's `api/<version>/<kind>_types.go`.
+
+In our example, `MemcachedSpec` and `MemcachedStatus` from [`pkg/apis/cache/v1aplha1/memcached_types.go`][memcached_types] are copied to `api/v1alpha1/memcached_types.go`.
+
+```go
+type MemcachedSpec struct {
+	// Specify size, the number of pods
+	Size int32 `json:"size"`
+}
+
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### CRD markers
+
+Include any of the CRD generation or validation markers defined for your API type and fields in `api/<version>/<kind>_types.go`.
+
+For `memcached-operator` the following markers present in `pkg/apis/cache/v1aplha1/memcached_types.go` are copied to `api/v1alpha1/memcached_types.go`.
+
+Example:
+```Go
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=memcacheds,scope=Namespaced
+```
+
+### Controller logic
+
+Copy over the `Reconcile()` code and the controller logic from your existing controller at `pkg/controller/<kind>/<kind>_controller.go` to the new controller's reconcile function in Kubebuilder project at `controllers/<kind>_controller.go`.
+
+In our example [`pkg/controller/memcached/memcached_controller.go`][memcached_controller] is copied to `controllers/memcached_controller.go`.
+
+Particularly, note the following changes:
+
+1. Update the import paths and names as necessary:
+
+```go
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	cachev1alpha1 "github.com/example-inc/memcached-operator/api/v1alpha1"
+)
+```
+
+2. The naming convention of the reconciler struct may differ. For example, in case of memcached operator the reconciler struct in SDK project is mentioned as [`ReconcileMemcached`][sdk_reconcile_struct] whereas in Kubebuilder project it would be `MemcachedReconciler`.
+
+```go
+type MemcachedReconciler struct {
+	Log    logr.Logger
+	Client client.Client
+	Scheme *runtime.Scheme
+}
+```
+
+3. Update the `Reconcile` logic and the related functions in `api/<version>/<kind>_controller.go`.
+
+For example, the reconcile logic in `api/v1alpha1/memcached_controller.go` would look like:
+
+```Go
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
```

Comment:
This example reconciler is probably going more into the user-guide side of things than the actual migration steps, so probably remove this entire reconciler block and instead point to the new example [memcached_controller.go.tmpl](https://github.com/operator-framework/operator-sdk/blob/master/example/kb-memcached-operator/memcached_controller.go.tmpl#L46-L122) file that was just added.